### DISCUSSION TITLE
When I call setActiveLang it trigger CD?

### DISCUSSION THREAD

**Discussion by jon9090** [ngneat/transloco#762] [Q&A]
When I call `setActiveLang` it trigger CD? if not, is it possible to `transloco` to trigger CD in a different way?

**Comment by shaharkazaz** [comment]
This is a general question, can you please share your use case?
Generally speaking, when you call [`setActiveLang`](https://github.com/jsverse/transloco/blob/master/libs/transloco/src/lib/transloco.service.ts#L159) it triggers a new value in some streams such as the `lang$`, let's say you are using the structural directive approach, so this directive [listens](https://github.com/jsverse/transloco/blob/master/libs/transloco/src/lib/transloco.directive.ts#L98) to this stream and in turn it will call [`cdr.markForCheck`](https://github.com/jsverse/transloco/blob/master/libs/transloco/src/lib/transloco.directive.ts#L128).
